spec/requests/conversion_hosts_spec.rb

```diff
@@ -70,50 +76,79 @@
+    end
+
     it "supports single conversion host creation" do
       api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
 
       post(api_conversion_hosts_url, :params => sample_conversion_host_from_vm)
 
       expect(response).to have_http_status(:ok)
-      expect_result_resources_to_include_keys("results", expected_attributes)
 
-      conversion_host_id = response.parsed_body["results"].first["id"]
+      results = response.parsed_body["results"].first
 
-      expect(ConversionHost.find_by(:resource_id => vm.id).id).to eql(conversion_host_id.to_i)
+      expect(results['success']).to be_truthy
+      expect(results['href']).to eql('http://www.example.com/api/conversion_hosts/')
+      expect(results['message']).to eql("Enabling resource id:#{vm.id} type:Vm")
+      expect(results['task_id']).to match(/\d+/)
+      expect(MiqTask.exists?(results['task_id'].to_i)).to be_truthy
     end
 
-    it "supports single conversion host creation via action" do
+    it "supports multiple conversion host creation" do
       api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
 
-      post(api_conversion_hosts_url, :params => gen_request(:create, sample_conversion_host_from_vm))
+      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
+      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
 
       expect(response).to have_http_status(:ok)
-      expect_result_resources_to_include_keys("results", expected_attributes)
 
-      conversion_host_id = response.parsed_body["results"].first["id"]
-      expect(ConversionHost.find_by(:resource_id => vm.id).id).to eql(conversion_host_id.to_i)
+      results = response.parsed_body["results"]
+
+      expect(results).to match_array([
+        a_hash_including("message" => "Enabling resource id:#{vm.id} type:Vm", "task_id" => a_kind_of(String)),
+        a_hash_including("message" => "Enabling resource id:#{host.id} type:Host", "task_id" => a_kind_of(String)),
+      ])
     end
+  end
 
-    it "supports multiple conversion host creation" do
-      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+  context "disable" do
+    let(:zone) { FactoryBot.create(:zone, :name => "api_zone") }
+    let(:ems) { FactoryBot.create(:ems_vmware, :zone => zone) }
+    let(:vm) { FactoryBot.create(:vm, :ems_id => ems.id) }
+    let(:conversion_host) { FactoryBot.create(:conversion_host, :resource => vm) }
+    let(:conversion_host_url) { api_conversion_host_url(nil, conversion_host) }
 
-      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
-      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
+    before do
+      allow(conversion_host).to receive(:install_conversion_host_module).and_return(true)
+      allow_any_instance_of(ConversionHost).to receive(:ansible_playbook).and_return({})
```

Comment:
how about mocking the `resource_search` to return the `conversion_host` and so you can mock that particular instance, instead of `allow_any_instance_of`?

Reply:
Done.